### PR TITLE
ISSUE-5019: re-reorganization the common crates

### PR DESCRIPTION
**Pull request by usamoi**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Merges `common-mem-allocator`, `common-containers` and `common-range-map` to `common-base`.

## Changelog

Not for changelog

## Related Issues

Closes https://github.com/datafuselabs/databend/issues/5019.

### CODE REVIEW

File: common/base/src/base/mod.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+mod format;
```
Comment (BohuTANG) [inline common/base/src/base/mod.rs:1]: It would be better to add the copyright :D

**Review by BohuTANG — COMMENTED** [review]
Please remove the `.idea` directory.

**Comment by usamoi** [comment]
@BohuTANG Sorry. Fixed now.

**Review by BohuTANG — APPROVED** [review]
A monster PR, love it to make the `common` crate cleaner.
Great work, thank you for the contribution, @usamoi

**Pull request merged**